lms/djangoapps/certificates/api.py

```diff
@@ -452,3 +452,41 @@ def emit_certificate_event(event_name, user, course_id, course=None, event_data=
 
     with tracker.get_tracker().context(event_name, context):
         tracker.emit(event_name, event_data)
+
+
+def get_certificate_header():
+    """
+    Return data to be used in Certificate Header,
+    data returned should be customized according to the microsite settings
+    """
+    data = dict(
+        logo_src=get_logo_url(),
+        logo_url=branding_api.get_base_url(),
+    )
+
+    return data
+
+
+def get_certificate_footer():
+    """
+    Return data to be used in Certificate Footer,
+    data returned should be customized according to the microsite settings
+    """
+    data = dict()
+
+    # get Terms of Service and Honor Code page url
+    terms_of_service_and_honor_code = branding_api.get_tos_and_honor_code_url()
+    if terms_of_service_and_honor_code != branding_api.EMPTY_URL:
```

Comment:
@saleem-latif @mattdrayer I'm trying to run the Open edX web certs locally and I'm getting an error because this template assumes that there is always a ``company_tos_url``:

https://github.com/edx/edx-platform/blob/master/lms/templates/certificates/_accomplishment-footer.html#L10

I assume that this is intended to continue working, so can this template be fixed to add the necessary conditional. It would also be good to add a test that doesn't pass this value, since that appears to be the default for a vanilla Open edX project.

FYI, here's the stack trace:

```
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/util/views.py", line 87, in inner
    return func(request, *args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/views/webview.py", line 550, in render_html_view
    return _render_certificate_template(request, context, course, user_certificate)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/views/webview.py", line 402, in _render_certificate_template
    return render_to_response("certificates/valid.html", context)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 170, in render_to_response
    return HttpResponse(render_to_string(template_name, dictionary, context_instance, namespace), **kwargs)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 160, in render_to_string
    return template.render_unicode(**context_dictionary)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/template.py", line 454, in render_unicode
    as_unicode=True)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 829, in _render
    **_kwargs_for_callable(callable_, data))
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 864, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 890, in _exec_template
    callable_(context, *args, **kwargs)
  File "/tmp/mako_lms/35b4de82addd2666f931e3dcd3a9da25/certificates/accomplishment-base.html.py", line 75, in render_body
    runtime._include_file(context, u'_accomplishment-footer.html', _template_uri)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 752, in _include_file
    callable_(ctx, **_kwargs_for_include(callable_, context._data, **kwargs))
  File "/tmp/mako_lms/35b4de82addd2666f931e3dcd3a9da25/certificates/_accomplishment-footer.html.py", line 29, in render_body
    __M_writer(filters.decode.utf8(company_tos_url))
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/filters.py", line 78, in decode
    return decode(str(x))
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 226, in __str__
    raise NameError("Undefined")
```